pkg/analysis/commentstart/analyzer.go

```diff
@@ -26,79 +25,37 @@ var Analyzer = &analysis.Analyzer{
 	Name:     name,
 	Doc:      "Check that all struct fields in an API have a godoc, and that the godoc starts with the serialised field name",
 	Run:      run,
-	Requires: []*analysis.Analyzer{inspect.Analyzer, extractjsontags.Analyzer},
+	Requires: []*analysis.Analyzer{inspector.Analyzer},
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
-	inspect, ok := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	inspect, ok := pass.ResultOf[inspector.Analyzer].(inspector.Inspector)
 	if !ok {
 		return nil, errCouldNotGetInspector
 	}
 
-	jsonTags, ok := pass.ResultOf[extractjsontags.Analyzer].(extractjsontags.StructFieldTags)
-	if !ok {
-		return nil, errCouldNotGetJSONTags
-	}
-
-	// Filter to structs so that we can iterate over fields in a struct.
-	nodeFilter := []ast.Node{
-		(*ast.Field)(nil),
-	}
-
-	inspect.WithStack(nodeFilter, func(n ast.Node, push bool, stack []ast.Node) (proceed bool) {
-		if !push {
-			return false
-		}
-
-		if len(stack) < 2 {
-			return true
-		}
-
-		// The 0th node in the stack is the *ast.File.
-		// The 1st node in the stack is the *ast.GenDecl.
-		decl, ok := stack[1].(*ast.GenDecl)
-		if !ok {
-			return false
-		}
-
-		if decl.Tok != token.TYPE {
-			// Returning false here means we won't inspect non-type declarations (e.g. var, const, import).
-			return false
-		}
```

Comment:
@sivchari I've dropped this if statement, I couldn't find, through my testing, any way to trigger this branch, so it became dead code.

Do you happen to have a specific example I could add to the tests that would trigger this branch?

Reply:
```go
var VarA = struct {
	Foo string
}{}
```

If the statement is dropped, this case can't catch.

Reply:
Added this back in, thank you!